### PR TITLE
Add error messages for BlockstoreError

### PR DESCRIPTION
**Pull request by KirillLykov**
#### Problem

When `BlockstoreError` is logged, it is impossible to see the value of the particular variant of the enum.

### CODE REVIEW

**Review by steviez — COMMENTED** [review]
For all of the variants that have a `#[from ...`, it'd be nice to display the underlying error as well. I think this can be done by adding `{0}`; see below for example:
https://github.com/solana-labs/solana/blob/3508b7d84ee3d05107c908ff2bb4772f3524e22e/ledger/src/blockstore_processor.rs#L590-L592

**Comment by KirillLykov** [comment]
> For all of the variants that have a `#[from ...`, it'd be nice to display the underlying error as well. I think this can be done by adding `{0}`; see below for example:
> 
> https://github.com/solana-labs/solana/blob/3508b7d84ee3d05107c908ff2bb4772f3524e22e/ledger/src/blockstore_processor.rs#L590-L592

done

File: ledger/src/blockstore_db.rs [hunk 1]
```diff
@@ -106,35 +106,49 @@ const OPTIMISTIC_SLOTS_CF: &str = "optimistic_slots";
 
 #[derive(Error, Debug)]
 pub enum BlockstoreError {
+    #[error("shred for index exists")]
     ShredForIndexExists,
+    #[error("invalid shred data")]
     InvalidShredData(Box<bincode::ErrorKind>),
+    #[error("RocksDB error, error: {0}")]
```
Comment (steviez) [inline ledger/src/blockstore_db.rs:113]:
```suggestion
    #[error("RocksDB error: {0}")]
```

File: ledger/src/blockstore_db.rs [hunk 2]
```diff
@@ -106,35 +106,49 @@ const OPTIMISTIC_SLOTS_CF: &str = "optimistic_slots";
...
     RocksDb(#[from] rocksdb::Error),
+    #[error("slot is not rooted")]
     SlotNotRooted,
+    #[error("dead slot")]
     DeadSlot,
+    #[error("io error, error: {0}")]
```
Comment (steviez) [inline ledger/src/blockstore_db.rs:119]:
```suggestion
    #[error("io error: {0}")]
```

File: ledger/src/blockstore_db.rs [hunk 3]
```diff
@@ -106,35 +106,49 @@ const OPTIMISTIC_SLOTS_CF: &str = "optimistic_slots";
...
     Io(#[from] std::io::Error),
+    #[error("serialization error, error: {0}")]
```
Comment (steviez) [inline ledger/src/blockstore_db.rs:121]:
```suggestion
    #[error("serialization error: {0}")]
```

File: ledger/src/blockstore_db.rs [hunk 4]
```diff
@@ -106,35 +106,49 @@ const OPTIMISTIC_SLOTS_CF: &str = "optimistic_slots";
...
     Serialize(#[from] Box<bincode::ErrorKind>),
+    #[error("fs extra error, error: {0}")]
```
Comment (steviez) [inline ledger/src/blockstore_db.rs:123]:
```suggestion
    #[error("fs extra error: {0}")]
```

File: ledger/src/blockstore_db.rs [hunk 5]
```diff
@@ -106,35 +106,49 @@ const OPTIMISTIC_SLOTS_CF: &str = "optimistic_slots";
...
     UnableToSetOpenFileDescriptorLimit,
+    #[error("trnasaction status slot mismatch")]
     TransactionStatusSlotMismatch,
+    #[error("empty epoch stakes")]
     EmptyEpochStakes,
+    #[error("no vote timestamps in range")]
     NoVoteTimestampsInRange,
+    #[error("protobuf encode error, error: {0}")]
     ProtobufEncodeError(#[from] prost::EncodeError),
+    #[error("protobuf decode error, error: {0}")]
```
Comment (steviez) [inline ledger/src/blockstore_db.rs:139]:
```suggestion
    #[error("protobuf decode error: {0}")]
```

File: ledger/src/blockstore_db.rs [hunk 6]
```diff
@@ -106,35 +106,49 @@ const OPTIMISTIC_SLOTS_CF: &str = "optimistic_slots";
...
     Serialize(#[from] Box<bincode::ErrorKind>),
+    #[error("fs extra error: {0}")]
     FsExtraError(#[from] fs_extra::error::Error),
+    #[error("slot cleaned up")]
     SlotCleanedUp,
+    #[error("unpack error: {0}")]
     UnpackError(#[from] UnpackError),
+    #[error("unable to set open file description limit")]
     UnableToSetOpenFileDescriptorLimit,
+    #[error("trnasaction status slot mismatch")]
```
Comment (steviez) [inline ledger/src/blockstore_db.rs:131]:
```suggestion
    #[error("transaction status slot mismatch")]
```

File: ledger/src/blockstore_db.rs [hunk 7]
```diff
@@ -106,35 +106,49 @@ const OPTIMISTIC_SLOTS_CF: &str = "optimistic_slots";
...
     Io(#[from] std::io::Error),
+    #[error("serialization error: {0}")]
     Serialize(#[from] Box<bincode::ErrorKind>),
+    #[error("fs extra error: {0}")]
     FsExtraError(#[from] fs_extra::error::Error),
+    #[error("slot cleaned up")]
     SlotCleanedUp,
+    #[error("unpack error: {0}")]
     UnpackError(#[from] UnpackError),
+    #[error("unable to set open file description limit")]
```
Comment (KirillLykov) [inline ledger/src/blockstore_db.rs:129]:
```suggestion
    #[error("unable to set open file descriptor limit")]
```

**Pull request merged**